bindings/wysiwyg-ffi/src/ffi_composer_update.rs

```diff
@@ -121,35 +121,118 @@ mod test {
+        let mut model = Arc::new(ComposerModel::new());
+        model.replace_text("Like  said".into());
+        model.select(5, 5); // "Like | said"
+
+        insert_mention_at_cursor(&mut model);
+
+        assert_eq!(
+            model.get_content_as_html(),
+            "Like <a data-mention-type=\"user\" href=\"https://matrix.to/#/@alice:matrix.org\" contenteditable=\"false\">Alice</a> said",
+        )
+    }
 
-        model.set_link_suggestion(
+    #[test]
+    fn test_replace_text_in_second_paragraph_node_with_mention_ffi() {
+        let mut model = Arc::new(ComposerModel::new());
+        model.replace_text("hello".into());
+        model.enter();
+        insert_mention_at_cursor(&mut model);
+
+        assert_eq!(
+            model.get_content_as_html(),
+            "<p>hello</p><p><a data-mention-type=\"user\" href=\"https://matrix.to/#/@alice:matrix.org\" contenteditable=\"false\">Alice</a>\u{a0}</p>",
+        )
+    }
+
+    #[test]
+    fn test_replace_text_in_second_list_item_start_with_mention_ffi() {
+        let mut model = Arc::new(ComposerModel::new());
+
+        model.ordered_list();
+        model.replace_text("hello".into());
+        model.enter();
+
+        insert_mention_at_cursor(&mut model);
+
+        assert_eq!(
+            model.get_content_as_html(),
+            "<ol><li>hello</li><li><a data-mention-type=\"user\" href=\"https://matrix.to/#/@alice:matrix.org\" contenteditable=\"false\">Alice</a>\u{a0}</li></ol>",
+        )
+    }
+
+    #[test]
+    fn test_replace_text_in_second_list_item_end_with_mention_ffi() {
+        let mut model = Arc::new(ComposerModel::new());
+        model.ordered_list();
+        model.replace_text("hello".into());
+        model.enter();
+        model.replace_text("there ".into());
+
+        insert_mention_at_cursor(&mut model);
+
+        assert_eq!(
+            model.get_content_as_html(),
+            "<ol><li>hello</li><li>there <a data-mention-type=\"user\" href=\"https://matrix.to/#/@alice:matrix.org\" contenteditable=\"false\">Alice</a>\u{a0}</li></ol>",
+        )
+    }
+
+    fn insert_mention_at_cursor(model: &mut Arc<ComposerModel>) {
+        let update = model.replace_text("@alic".into());
+        let MenuAction::Suggestion{suggestion_pattern} = update.menu_action() else {
+        panic!("No suggestion pattern found")
+    };
+        model.insert_mention_at_suggestion(
             "https://matrix.to/#/@alice:matrix.org".into(),
             "Alice".into(),
             suggestion_pattern,
-            vec![
-                Attribute {
-                    key: "contenteditable".into(),
-                    value: "false".into(),
-                },
-                Attribute {
-                    key: "data-mention-type".into(),
-                    value: "user".into(),
-                },
-            ],
+            vec![Attribute {
+                key: "data-mention-type".into(),
+                value: "user".into(),
+            }],
```

Comment:
We should soon be able to remove this so could add a TODO here (issue https://github.com/matrix-org/matrix-rich-text-editor/issues/709)